src/decision/test/vision-decision-test.cpp

```diff
@@ -6,10 +6,151 @@
 
 #include <VisionDecision.h>
 #include <gtest/gtest.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <cv_bridge/cv_bridge.h>
+#include <tiff.h>
 
+using namespace cv;
 
+sensor_msgs::Image convertToSensorMsg(Mat cvMatImage);
 
-int main(int argc, char **argv) {
-    testing::InitGoogleTest(&argc, argv);
+TEST(imageTest, angleStraight){
+    String filename = "imageTests/testStraightImage.jpg";
+    Mat image = imread(filename, CV_LOAD_IMAGE_GRAYSCALE);
+
+    sensor_msgs::Image sensorMsg = convertToSensorMsg(image);
+
+    sensor_msgs::ImageConstPtr testImageScan(new sensor_msgs::Image(sensorMsg));
+
+    EXPECT_EQ(0, VisionDecision::getDesiredAngle(300, testImageScan));
+}
+
+TEST(imageTest, angleLeft){
+    String filename = "imageTests/testLeftImage.jpg";
+    Mat image = imread(filename, CV_LOAD_IMAGE_GRAYSCALE);
+
+    sensor_msgs::Image sensorMsg = convertToSensorMsg(image);
+
+    sensor_msgs::ImageConstPtr testImageScan(new sensor_msgs::Image(sensorMsg));
+
+    EXPECT_EQ(-37, VisionDecision::getDesiredAngle(300, testImageScan));
+}
+
+TEST(imageTest, angleRight){
+    String filename = "imageTests/testNoisyRightImage.jpg";
+    Mat image = imread(filename, CV_LOAD_IMAGE_GRAYSCALE);
+
+    sensor_msgs::Image sensorMsg = convertToSensorMsg(image);
+
+    sensor_msgs::ImageConstPtr testImageScan(new sensor_msgs::Image(sensorMsg));
+
+    EXPECT_EQ(33, VisionDecision::getDesiredAngle(300, testImageScan));
+}
+
+TEST(imageTest, noisyStraight){
+    String filename = "imageTests/testVeryNoisyStraightImage.jpg";
+    Mat image = imread(filename, CV_LOAD_IMAGE_GRAYSCALE);
+
+    sensor_msgs::Image sensorMsg = convertToSensorMsg(image);
+
+    sensor_msgs::ImageConstPtr testImageScan(new sensor_msgs::Image(sensorMsg));
+
+    EXPECT_EQ(-5, VisionDecision::getDesiredAngle(300, testImageScan));
+}
+
+TEST(imageTest, noisyLeft){
+    String filename = "imageTests/testVeryNoisyLeftImage.jpg";
+    Mat image = imread(filename, CV_LOAD_IMAGE_GRAYSCALE);
+
+    sensor_msgs::Image sensorMsg = convertToSensorMsg(image);
+
+    sensor_msgs::ImageConstPtr testImageScan(new sensor_msgs::Image(sensorMsg));
+
+    EXPECT_EQ(-37, VisionDecision::getDesiredAngle(300, testImageScan));
+}
+
+TEST(imageTest, splitLines){
```

Comment:
Not sure what "split" means throughout this file. Can you add a comment to each test with "split" in the title quickly explaining what it does? Doesn't have to be a full Javadoc comment, a quick one-liner should be fine